plugins/catalog-backend-module-backstage/src/module.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  coreServices,
+  createBackendModule,
+} from '@backstage/backend-plugin-api';
+import { loggerToWinstonLogger } from '@backstage/backend-common';
+import { catalogProcessingExtensionPoint } from '@backstage/plugin-catalog-node/alpha';
+import { BackstageEntityProvider } from './provider/BackstageEntityProvider';
+
+export const catalogModuleBackstageEntityProvider = createBackendModule({
+  pluginId: 'catalog',
+  moduleId: 'backstage-entity-provider',
+  register(env) {
+    env.registerInit({
+      deps: {
+        catalog: catalogProcessingExtensionPoint,
+        config: coreServices.rootConfig,
+        logger: coreServices.logger,
+        scheduler: coreServices.scheduler,
+        urlReader: coreServices.urlReader,
+      },
+      async init({ catalog, config, logger, scheduler, urlReader }) {
+        catalog.addEntityProvider(
+          BackstageEntityProvider.fromConfig(config, {
+            logger: loggerToWinstonLogger(logger),
```

Comment:
canonical thing is to make the class accept `LoggerService`, `ConfigService` etc in the first place instead of a winston logger and pure `Config` etc etc, unless you specifically want to be able to use the class from the old backend system

Reply:
👍